.github/actions/is-pr-approved/src/main.ts

```diff
@@ -0,0 +1,161 @@
+import * as core from '@actions/core'
+import {Context} from '@actions/github/lib/context'
+import {GitHub} from '@actions/github/lib/utils'
+import {actionInputs, transformIfSet} from 'github-actions-utils'
+
+interface GitHubAPIError {
+  status: number
+  request: any // eslint-disable-line @typescript-eslint/no-explicit-any
+  response: any // eslint-disable-line @typescript-eslint/no-explicit-any
+}
+
+export const DEFAULT_ALLOWED_ACTORS: readonly string[] = [
+  'greenkeeper[bot]',
+  'dependabot[bot]',
+  'mergify[bot]',
+  'github-actions[bot]'
+]
+
+export async function run(): Promise<void> {
+  // Use dynamic import to let `github-action-ts-run-api` library inject
+  // Github Context before using it.
+  // Static import will not work and result in empty Github Context
+  const github = await import('@actions/github')
+
+  try {
+    // Validate event source
+    if (!['pull_request', 'pull_request_target'].includes(github.context.eventName)) {
+      return core.setFailed('This action only support `pull_request` and `pull_request_target` events')
```

Comment:
Given the below GitHub action:

```yaml

on:
  workflow_dispatch:
  pull_request:
  push:
    branches: [ 'main']
...
```

Will it fail when running a manual build using the GitHub UI or when merging into `main`?

Reply:
Currently, both will fail due to the specific PR review logic.

Reply:
I added support for those events. Do we need to rename the action to something like `is-workflow-approved` to reflect the support of those events?